core/lib/fields/richText.ts

```diff
@@ -8,3 +14,100 @@ export const serializeProseMirrorDoc = (value: Node) => {
 	// We stringify and then parse in order to make it a plain JSON object
 	return JSON.parse(JSON.stringify(value.toJSON()));
 };
+
+interface GetPubValuesReturnType {
+	[pubId: string]: {
+		parentPubId: string;
+		pubId: string;
+		pubTypeId: string;
+		values: {
+			[fieldSlug: string]: {
+				type: string;
+				attrs: {
+					id: string | null;
+					class: string | null;
+				};
+				content: {
+					type: string;
+					text: string;
+				}[];
+			}[];
+		};
+	};
+}
+
+interface PubCreate {
+	id: string;
+	pubTypeId: string;
+	values: {
+		[fieldSlug: string]: any;
+	};
+}
+
+/**
+ * Parse rich text values to get values (overwritten pubfields) and children (related pubs)
+ * in order to pass into a `createPub` function.
+ *
+ * If an editor has something like
+ *
+ * paragraph
+ * _______
+ * | My content here
+ *
+ * /Submission (pub type)
+ * _______
+ * |
+ *
+ * /Title (pub field)
+ * _______
+ * | My new title
+ *
+ * This would return a new "Submission" pub with empty fields in children,
+ * and a new title field in values (i.e. {"croccroc:title":"My new title" })
+ *
+ * TODO: can the paragraph content "My content here" just stay in the RichText field?
+ * TODO: we can't fill out the Submission pub type without ContextAtom rendering first
+ */
+export const parseRichTextForPubFieldsAndPubChildren = ({
```

Comment:
this function is meant to be called by a form that has a rich text field (i.e. `ExternalFormWrapper` or `PubEditorClient`) before sending to the `createPubRecursiveNew` action. It attempts to pull out the pubfields and children pub using the editor's `getPubValues` function, and put that into a form that `createPubRecursiveNew` can take